XenModel/Actions/Pool_Patch/GetDiskSpaceRequirementsAction.cs

```diff
@@ -62,7 +63,16 @@ public GetDiskSpaceRequirementsAction(Host host, Pool_patch patch, bool suppress
         /// </summary>
         public GetDiskSpaceRequirementsAction(Host host, string path, bool suppressHistory)
             : this(host, FileName(path), FileSize(path), suppressHistory)
-        { }
+        { }
+
+        /// <summary>
+        /// This constructor is used to calculate the disk space when the required space is known only
+        /// </summary>
+        public GetDiskSpaceRequirementsAction(Host host, long size, bool suppressHistory)

```

Comment:
I think it would be better to pass the operation type as a parameter to this constructor, because it is not clear from its description or the signature that the operation it's going to be set to autoupdate.